llmfoundry/callbacks/curriculum_learning_callback.py

```diff
@@ -34,20 +40,189 @@ class CurriculumLearning(CallbackWithConfig):
         dataset_index (int): The index of the dataset currently being used.
     """
 
-    def __init__(self, train_config: Dict, dataset_index: int):
-        self.dataset_index = dataset_index
-        self.saved_dataset_index = 0
-        self.all_dataset_configs = []
-        self.current_dataset_state = {}
-        # The current dataset config is resolved and passed in train.py
-        self.current_dataset_config = train_config['train_loader']
+    def __init__(
+        self,
+        train_config: dict[str, Any],
```

Comment:
is this still necessary? since now the callback stores the entire state of all phases

Reply:
This is needed to get the tokenizer config to build the tokenizer (which is explained more in the comment below).